### PR TITLE
Islandpath

### LINKED ISSUE #3142: new module: IslandPath
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jvfe**
Adds a module to run [IslandPath](https://github.com/brinkmanlab/islandpath), a software for genomic island detection in annotated genomes.

I had to use GUNZIP inside of the IslandPath test because it doesn't allow for compressed input files.

Conda runs tend to be very unstable, so I thought to add this to the conda exclude list in the CI, but let's see how this goes first.

## PR checklist

Closes #3142

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by louperelo — COMMENTED** [review]
Oi João,
Nice work. Looks like you have to disable Conda in the CI tests, I tried to re-run the test and it failed again. Did it work locally?

File: modules/nf-core/islandpath/meta.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - genome:
+      type: file
+      description: |
+        Genome file in .gbk or .embl format.
```
[the remark this answers is not in the document]
Comment (jvfe) [reply modules/nf-core/islandpath/meta.yml:28]: Yep, that makes sense, thank you!

**Comment by jvfe** [comment]
> Oi João, Nice work. Looks like you have to disable Conda in the CI tests, I tried to re-run the test and it failed again. Did it work locally?

It fails more often than it works with conda 🤣 . I've added it to the conda exclude list in the CI. Thank you for your review!

**Comment by louperelo** [comment]
@nf-core-bot fix linting

**Pull request merged**